staging/src/k8s.io/kubectl/pkg/cmd/clusterinfo/clusterinfo_dump.go

```diff
@@ -154,12 +154,12 @@ func (o *ClusterInfoDumpOptions) Complete(f cmdutil.Factory, cmd *cobra.Command)
 		return err
 	}
 
-	o.Namespace, _, err = f.ToRawKubeConfigLoader().Namespace()
+	o.Namespace, _, err = restClientGetter.ToRawKubeConfigLoader().Namespace()
 	if err != nil {
 		return err
 	}
 	// TODO this should eventually just be the completed kubeconfigflag struct
```

Comment:
Probably, it is good time also to handle this `TODO` but I don't fully understand what this means

Reply:
If you look at ConfigFlags https://github.com/kubernetes/kubernetes/blob/a837be03ffea65115a90b4bc4a8092d9b842971e/staging/src/k8s.io/cli-runtime/pkg/genericclioptions/config_flags.go#L78 they implement `RESTClientGetter`. Currently we inject `Factory` still https://github.com/kubernetes/kubernetes/blob/a837be03ffea65115a90b4bc4a8092d9b842971e/staging/src/k8s.io/kubectl/pkg/cmd/cmd.go#L324 into every command, but the goal will be to entirely get rid of that, and read the flags directly within each command to be able to treat each command as fully autonomous. You can safely drop that TODO, since it'll be a while since we get there :wink: